src/features/command-palette/command-palette.ts

```diff
@@ -0,0 +1,180 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { AutocompleteResult, AutocompleteSettings } from "autocompleter";
+import { inject, injectable } from "inversify";
+import { Action, isAction } from "../../base/actions/action";
+import { IActionDispatcherProvider } from "../../base/actions/action-dispatcher";
+import { CommandExecutionContext } from "../../base/commands/command";
+import { SModelElement } from "../../base/model/smodel";
+import { findParentByFeature } from "../../base/model/smodel-utils";
+import { TYPES } from "../../base/types";
+import { AbstractUIExtension } from "../../base/ui-extensions/ui-extension";
+import { HideUIExtensionAction, ShowUIExtensionAction } from "../../base/ui-extensions/ui-extension-registry";
+import { KeyListener } from "../../base/views/key-tool";
+import { ViewerOptions } from "../../base/views/viewer-options";
+import { toArray } from "../../utils/iterable";
+import { matchesKeystroke } from "../../utils/keyboard";
+import { ILogger } from "../../utils/logging";
+import { isBoundsAware } from "../bounds/model";
+import { isSelectable } from "../select/model";
+import { isViewport } from "../viewport/model";
+import { ICommandPaletteActionProviderRegistry, LabeledAction, isLabeledAction } from "./action-providers";
+
+
+// import of function autocomplete(...) doesn't work
+// see also https://github.com/kraaden/autocomplete/issues/13
+// this is a workaround to still get the function including type support
+const configureAutocomplete: (settings: AutocompleteSettings<LabeledAction>) => AutocompleteResult = require("autocompleter");
+
+@injectable()
+export class CommandPalette extends AbstractUIExtension {
+    static readonly ID = "command-palette";
+    readonly id = CommandPalette.ID;
+
+    readonly containerClass = "command-palette";
+    readonly xOffset = 20;
+    readonly yOffset = 30;
+    readonly defaultWidth = 400;
+
+    protected inputElement: HTMLInputElement;
+    protected autoCompleteResult: AutocompleteResult;
+
+    protected paletteContext?: CommandExecutionContext;
+    protected contextActions?: LabeledAction[];
+
+    constructor(
+        @inject(TYPES.ViewerOptions) protected options: ViewerOptions,
+        @inject(TYPES.IActionDispatcherProvider) protected actionDispatcherProvider: IActionDispatcherProvider,
+        @inject(TYPES.ICommandPaletteActionProviderRegistry) protected actionProvider: ICommandPaletteActionProviderRegistry,
+        @inject(TYPES.ILogger) protected logger: ILogger) {
+        super(options, logger);
```

Comment:
Please consider using property injection instead of constructor injection (see #59)